src/rtc.c

```diff
@@ -0,0 +1,518 @@
+/**
+ * @file rtc.c
+ * @brief Real-Time Clock Subsystem
+ * @ingroup rtc
+ */
+
+#include "libdragon.h"
+
+/**
+ * @defgroup rtc Real-Time Clock Subsystem
+ * @ingroup libdragon
+ * @brief Joybus real-time clock interface.
+ *
+ * The Joybus real-time clock is an accessory that was only included on
+ * one official cartridge that was only available in Japan. The Joybus RTC
+ * is accessed through the serial interface (SI) similar to EEPROM.
+ * A real RTC uses a battery to power a clock that runs even when the N64
+ * is powered-off and maintains the date, time, and day of the week.
+ *
+ * To check if the real-time clock is available, call #joybus_rtc_is_present.
+ * If detected, it is a good idea to ensure the RTC control block is in a
+ * known-good state by calling the #joybus_rtc_run function before reading
+ * the current time with #joybus_rtc_read_time for the first time.
+ *
+ * The simplest method to change the time is by calling #joybus_rtc_set,
+ * which is a high-level convenience function that prepares the RTC and
+ * writes the time with the proper delays.
+ *
+ * In general, since only one game ever used Joybus RTC (and that game was
+ * later re-released on the GameCube in English), RTC support by emulators
+ * and flash carts tends to be incomplete, inaccurate, or non-existent.
+ * Many emulators do not actually implement the RTC write command and will
+ * always respond with the host system's current local time. Some emulators
+ * and flash carts support writing to RTC but will not persist the date/time
+ * after resetting or powering-off.
+ *
+ * The only way to check if writes are actually supported is to write a
+ * time to the RTC and read the time back out. Many emulators that do
+ * support RTC reads will silently ignore RTC writes. You should detect
+ * whether writes are supported using #joybus_rtc_is_writable so that you can
+ * conditionally show the option to change the time if it's supported.
+ *
+ * The RTC sends and receives the date/time in bytes of binary-coded decimal
+ * numbers. This subsystem handles decoding and encoding the date/time into
+ * a more generally-useful struct called #rtc_time_t.
```

Comment:
The internal protocol is not very useful in this overview, so I'd remove this, though I'm not strong on this. Up to you.

Reply:
I mentioned it here because I want this to be a resource for people to understand how it actually works as well as how to use the APIs that we provide.

I don't think it's hurting anything to keep it, and I think it helps contextualize what the library is actually doing for them.

Reply:
ok